lifting-state-up/passing-down-event-handlers.md

```diff
@@ -0,0 +1,250 @@
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const updateStudentData = `                        | We're creating a function named `updateStudentData`.                                                                                                                 |
+| `updatedStudent`                                    | This function accepts one argument: an object that holds updated student data. We expect this object to have the key `id`, at the very least.                        |
+| `const students = ;`                                | For this function, we'll create a helper array, `students` to contain the updated student data.                                                                      |
+| `studentData.map(...)`                              | We'll map over `studentData`...                                                                                                                                      |
+| `student => {...}`                                 | Inside `map`, we'll reference each item with `student`...                                                                                                            |
+| `if (student.id === updatedStudent.id) {`           | If we found the student that our `updatedStudent` is updating...                                                                                                     |
+| `return updatedStudent`                             | The item we should map to our `students` array is the updated student data!                                                                                          |
+| `else { return student; }`                          | Otherwise, we should map the unchanged `student` into `students`.                                                                                                    |
+| `setStudentData(students);`                         | Ultimately, we want to update the `studentData` on state. We use our state update function, `setStudentData`, and we update it to our newly formed `students` array. |
+
+Our `App` now defines and holds a function that can update our `studentData`, named `updateStudentData`.
+
+## Send This Function to `StudentList`
+
+Now that we've defined `updateStudentData`, imagine if other components could _use_ this function.
+
+If other components could use this function, they'd have the ability to update the `studentData` in `App`! We want _the `Student` component_ to update the `studentData` in `App`.
+
+Other components _can_ use the `updateStudentData` function if they can reference it... and they can, _using props_.
```

Comment:
this extra sentence might sound too redundant, but for some reason I feel like reiterating this (including reference as well)
```suggestion
Other components _can_ use the `updateStudentData` function if they can reference it... and they can, _using props_. We can pass a reference to the `updateStudentData` down to a component through its `props`!
```